src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Clamp.cs

```diff
@@ -0,0 +1,321 @@
+        /// <remarks>
+        /// <para>
+        /// This method effectively computes <c><paramref name="destination" />[i] = T.Clamp(<paramref name="x" />, <paramref name="min" />, <paramref name="max" />[i])</c>.
+        /// </para>
+        /// </remarks>
+        public static void Clamp<T>(T x, T min, ReadOnlySpan<T> max, Span<T> destination)
+            where T : INumber<T> =>
+            InvokeSpanScalarScalarIntoSpan<T, ClampOperatorMaxXMin<T>>(max, x, min, destination);
+
+        /// <summary>T.Clamp(x, min, max)</summary>
+        internal readonly struct ClampOperatorXMinMax<T> : ITernaryOperator<T>
+            where T : INumber<T>
+        {
+            public static bool Vectorizable => true;
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static T Invoke(T x, T min, T max) => T.Clamp(x, min, max);
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> min, Vector128<T> max)
+            {
+                if (Vector128.GreaterThanAny(min, max))
+                {
+                    ThrowHelper.ThrowArgument_MinGreaterThanMax();
+                }
+
+                return MinOperator<T>.Invoke(MaxOperator<T>.Invoke(x, min), max);
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector256<T> Invoke(Vector256<T> x, Vector256<T> min, Vector256<T> max)
+            {
+                if (Vector256.GreaterThanAny(min, max))
+                {
+                    ThrowHelper.ThrowArgument_MinGreaterThanMax();
+                }
+
+                return MinOperator<T>.Invoke(MaxOperator<T>.Invoke(x, min), max);
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> min, Vector512<T> max)
+            {
+                if (Vector512.GreaterThanAny(min, max))
+                {
+                    ThrowHelper.ThrowArgument_MinGreaterThanMax();
+                }
+
+                return MinOperator<T>.Invoke(MaxOperator<T>.Invoke(x, min), max);
+            }
+        }
+
+        /// <summary>T.Clamp(x, min, max) without per vector validation of min &lt;= max</summary>
+        internal readonly struct ClampOperatorXMinMaxNoValidation<T> : ITernaryOperator<T>
+            where T : INumber<T>
+        {
+            public static bool Vectorizable => true;
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static T Invoke(T x, T min, T max) =>
+                T.Min(T.Max(x, min), max);
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> min, Vector128<T> max) =>
+                MinOperator<T>.Invoke(MaxOperator<T>.Invoke(x, min), max);
```

Comment:
We should probably prefer a `ClampOperator<T>.Invoke` here which can use `Vector128.Clamp` as that can be more efficient where available.

Reply:
Ok